src/workflow/CommandExecutor.py

```
@@ -157,27 +167,30 @@ def read_stdout():
+            """Read stderr in real-time, buffering for conditional minimal log output"""
             try:
                 for line in iter(process.stderr.readline, ''):
                     if line:
-                        self.logger.log(f"STDERR: {line.rstrip()}", 0)
+                        stderr_line = line.rstrip()
+                        stderr_buffer.append(stderr_line)
+                        # Log to detailed log only during execution
+                        self.logger.log(f"STDERR: {stderr_line}", 2)
                     if process.poll() is not None:
                         break
             except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)